### ISSUE TITLE
about interleave labeled and unlabed samples

### ISSUE THREAD

**Issue by blankWorld** [YU1ut/MixMatch-pytorch#5]
Thanks for your implementation ! Why interleave labeled and unlabed samples between batches can get correct batchnorm calculation.

**Comment by Wondersui** [comment]
I have the same question

**Comment by YU1ut** [comment]
Batchnorm estimates the mean and std of the whole dataset by each batch. Since we feed labeled batch and unlabeled batch containing the same number of samples alternately, this estimation may be biased when the number of labeled samples is very small.

**Comment by blankWorld** [comment]
You are right, thanks for your answers, and I have another question. In one iteration of training, batchnorm layers are updated three times and each update is completed with 64 images. Why not update batchnorm layers one time using total 64*3 images in one iteration of training？

**Comment by YU1ut** [comment]
Yes, you can do that. But the learning rate should be adjusted according to the batch size. I just try to follow the official implementation in this repository.

**Comment by blankWorld** [comment]
Thanks again for your kindness. I have already reproducted  the official results.

**Comment by JongMokKim** [comment]
Hi blankWorld, did u make any change code to reproduce official result? or did u get with this code? Thank u in advance!
